xml/customFields_install.xml

```diff
@@ -0,0 +1,108 @@
+<?xml version="1.0" encoding="iso-8859-1" ?>
+<CustomData>
+  <CustomGroups>
+    <CustomGroup>
+      <name>external_direct_debit_mandate_information</name>
+      <title>External Direct Debit Mandate information</title>
+      <extends>ContributionRecur</extends>
+      <style>Inline</style>
+      <collapse_display>0</collapse_display>
+      <weight>100</weight>
+      <is_active>1</is_active>
+      <table_name>civicrm_value_external_dd_mandate_information</table_name>
+      <is_multiple>0</is_multiple>
+      <collapse_adv_display>0</collapse_adv_display>
+      <is_reserved>0</is_reserved>
+    </CustomGroup>
+  </CustomGroups>
+  <CustomFields>
+    <CustomField>
+      <name>mandate_id</name>
+      <label>Mandate Id</label>
+      <data_type>String</data_type>
+      <html_type>Text</html_type>
+      <is_required>0</is_required>
+      <is_searchable>1</is_searchable>
+      <is_search_range>0</is_search_range>
+      <weight>1</weight>
+      <is_active>1</is_active>
+      <is_view>0</is_view>
```

Comment:
@omarabuhussein  Shouldn't mandate_id, mandate_status, next_available_payment_date are read only?   So       `<is_view>0</is_view>` should be set to TRUE as these values should only be setup from GoCardless only.

Reply:
I think mandate id could be editable so we have a manual way to edit which mandate the recurring contribute is connected to.

And perhaps the others could be editable in order to deal with situations where civicrm gets out of sync?

Thoughts?

@erawat @omarabuhussein

Reply:
@erawat @jamienovick 

I allowed them to be edited intentionally , because as Jamie mentioned we need to allow users to deal with situations when such fields gets out of sync or if they want to override them manually in case something happens.

Reply:
@jamienovick  @omarabuhussein if that is the case, I am okay with making these fields edible (for now), but we need to consider that they are requirements from GoCardless that we need to handle mandate as below:

- displaying real-time mandate information to your users
- providing visibility to your users within your platform when an end customer (payer) cancels their mandate
- providing visibility to your users within your platform when a mandate activation fails
- providing visibility to your users within your platform when a mandate expires


Note: I am planing migrating the current subscription (donatioN) to have mandate sync with recurring contribution and this mandate should not allow to edit but again. 
We may need to be able to edit it if it is out of sync with a special permission, perhaps?